File: sdk/python/src/bsdkrun/images.py
"""Host-level image operations."""

from __future__ import annotations

import builtins
import json

from .process import run_checked
from .types import ImageInfo

__all__ = ["list"]


def list() -> builtins.list[ImageInfo]:
    """List downloaded images (pulled OCI images + fetched BSD images)."""
    res = run_checked(["images", "--json"], "bsdkrun images")
    rows = json.loads(res.stdout or "[]")
    return [ImageInfo.from_row(row) for row in rows]

File: sdk/ruby/lib/bsdkrun/images.rb
# frozen_string_literal: true

require "json"

module Bsdkrun
  # Host-level image operations.
  module Images
    module_function

    # List downloaded images (pulled OCI images + fetched BSD images).
    # @return [Array<ImageInfo>]
    def list
      res = Process.run!(["images", "--json"], label: "bsdkrun images")
      rows = JSON.parse(res.stdout.empty? ? "[]" : res.stdout)
      rows.map { |row| ImageInfo.from_row(row) }
    end
  end
end
